components/bsx/Offer/StatsOverview.vue

```diff
@@ -36,47 +36,41 @@
   </div>
 </template>
 
-<script lang="ts">
-import { Component, mixins } from 'nuxt-property-decorator'
+<script lang="ts" setup>
 import { OfferStats, StatsResponse } from './types'
-import PrefixMixin from '~/utils/mixins/prefixMixin'
 import { countOf } from '~/utils/countOf'
 import statsForBsx from '~/queries/subsquid/bsx/statsForBsx.graphql'
+import CommonTokenMoney from '@/components/shared/CommonTokenMoney.vue'
 
-const components = {
-  Identity: () => import('@/components/identity/IdentityIndex.vue'),
-  CommonTokenMoney: () => import('@/components/shared/CommonTokenMoney.vue'),
-}
+const { $apollo, $consola } = useNuxtApp()
+const { client } = usePrefix()
 
-@Component({ components })
-export default class BsxStats extends mixins(PrefixMixin) {
-  public statsResponse!: StatsResponse
-  protected offerStats: OfferStats[] = []
-  protected keysObject: string[] = []
+const statsResponse = ref<StatsResponse>()
+const offerStats = ref<OfferStats>()
+const keysObject = ref([])
 
-  fetch() {
-    this.getOfferStats()
+const returnTotalCounts = (key, statsResponse) => {
+  if (key === 'activeWallets') {
+    return countOf(statsResponse.activeWallets[0])
+  } else {
+    return countOf(statsResponse[key])
   }
+}
 
-  protected async getOfferStats() {
-    const response = await this.$apollo.query<StatsResponse>({
-      client: this.urlPrefix,
+useLazyAsyncData('data', async () => {
```

Comment:
this component didn't load on my end, unless use `watchEffect`

```suggestion
watchEffect(async () => {
```

Reply:
Oooooh this is clearly what I was looking for!!! I spent some time trying to know why it wasn't showing on fist load.
Do you know why component isn't triggered without? maybe because root component has already `useAsyncData`? Any Doc about that?

Reply:
Oh, sorry, I didn't see you are using `useLazyAsyncData` in this file also https://github.com/roiLeo/nft-gallery/blob/chore/stats/composition/components/bsx/Offer/OfferStats.vue#L33

It didn't load because both components used the same key `'data'`. The first parameter in `useAsyncData` should unique key https://nuxt.com/docs/api/composables/use-async-data#params

Reply:
Thank you for this explanation!